### PR TITLE
Add USM debugger

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Adds a minimal, self-contained build of USM for faster iterations of debugging eBPF code on remote machines. See the README for usage instructions.

### Motivation

To make development and testing easier.

### Describe how you validated your changes

Manual build using instructions and run of binary on some staging hosts.

### CODE REVIEW

File: pkg/network/usm/debugger/cmd/ebpf_bytecode.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	_ "embed"
+	"os"
+	"path"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// The code beloe is essentially responsible for embedding the CO-RE artifacts
```
Comment (gjulianm) [inline pkg/network/usm/debugger/cmd/ebpf_bytecode.go:18]:
```suggestion
// The code below is essentially responsible for embedding the CO-RE artifacts
```

**Review by guyarb — APPROVED** [review]

File: pkg/network/usm/debugger/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# USM debugger
+
+A minimal, self-contained build of USM for faster iterations of debugging eBPF
+code on remote machines.
+
+Prepare the build for the target architectures you need (the `foo` KMT stack
```
Comment (gjulianm) [inline pkg/network/usm/debugger/README.md:6]: I'd add a comment here that you need this for the eBPF object files, specially with cross-arch builds.

**Review by gjulianm — APPROVED** [review]

File: pkg/network/usm/debugger/README.md (hunk 2 above)
Comment (vitkyrka) [reply pkg/network/usm/debugger/README.md:6]: Done.

**Pull request merged**